fedimint-core/src/db/mod.rs

```diff
@@ -555,79 +561,89 @@ impl<'isolated, T: Send + Encodable> ModuleDatabaseTransaction<'isolated, T> {
             .await
             .expect("Error doing prefix search in database")
             .map(move |(key_bytes, value_bytes)| {
-                let key = KP::Record::from_bytes(&key_bytes, &decoders)?;
-                let value = decode_value(&value_bytes, &decoders)?;
-                Ok((key, value))
+                let key = KP::Record::from_bytes(&key_bytes, &decoders)
+                    .expect("Unrecoverable error reading the DatabaseKey");
+                let value = decode_value(&value_bytes, &decoders)
+                    .expect("Unrecoverable error decoding the DatabaseValue");
+                (key, value)
             })
     }
 
     #[instrument(level = "debug", skip_all, fields(?key, ?value), ret)]
-    pub async fn insert_entry<K>(&mut self, key: &K, value: &K::Value) -> Result<Option<K::Value>>
+    pub async fn insert_entry<K>(&mut self, key: &K, value: &K::Value) -> Option<K::Value>
     where
         K: DatabaseKey + DatabaseRecord,
     {
         self.commit_tracker.has_writes = true;
         match self
             .isolated_tx
             .raw_insert_bytes(&key.to_bytes(), value.to_bytes())
-            .await?
+            .await
+            .expect("Unrecoverable error while inserting into the database")
         {
-            Some(old_val_bytes) => Ok(Some(decode_value(&old_val_bytes, self.decoders)?)),
-            None => Ok(None),
+            Some(old_val_bytes) => Some(
+                decode_value(&old_val_bytes, self.decoders)
+                    .expect("Unrecoverable error while decoding the database value"),
+            ),
+            None => None,
         }
     }
 
     #[instrument(level = "debug", skip_all, fields(?key, ?value), ret)]
-    pub async fn insert_new_entry<K>(
-        &mut self,
-        key: &K,
-        value: &K::Value,
-    ) -> Result<Option<K::Value>>
+    pub async fn insert_new_entry<K>(&mut self, key: &K, value: &K::Value) -> Option<K::Value>
     where
         K: DatabaseKey + DatabaseRecord,
     {
         self.commit_tracker.has_writes = true;
         match self
             .isolated_tx
             .raw_insert_bytes(&key.to_bytes(), value.to_bytes())
-            .await?
+            .await
+            .expect("Unrecoverable error occurred while inserting new entry into database")
         {
             Some(_) => {
                 warn!(
                     target: LOG_DB,
                     "Database overwriting element when expecting insertion of new entry. Key: {:?}",
                     key
                 );
-                Ok(None)
+                None
             }
-            None => Ok(None),
+            None => None,
         }
```

Comment:
This could just be `if let Some(_) = ...`

Also, if we're always returning `None` do we even need a return value?

Reply:
Good catch, removed return value